### ISSUE TITLE
Destroy specific resource

### ISSUE THREAD

**Issue by Lewiscowles1986** [ljfranklin/terraform-resource#99]
I'm trying to use this to destroy a resource.

```
  - put: aws-sqs
    get_params:
      action: destroy module.app_name_queue
    params:
      <<: *aws-sqs-params
      action: destroy module.app_name_queue
```

This is my attempt to follow https://www.terraform.io/docs/commands/state/rm.html#example-remove-a-module

**Comment by ljfranklin** [comment]
@Lewiscowles1986 the `action` param doesn't accept multiple arguments, the only valid value is "destroy" to delete all resources in that environment's statefile. The resource doesn't provide any mechanism to delete individual resources as you almost never need to do that. From the page you linked:

```
Items removed from the Terraform state are not physically destroyed. Items removed from the Terraform state are only no longer managed by Terraform. For example, if you remove an AWS instance from the state, the AWS instance will continue running, but terraform plan will no longer see that instance.

There are various use cases for removing items from a Terraform state file. The most common is refactoring a configuration to no longer manage that resource (perhaps moving it to another Terraform configuration/state).
```

I'm assuming you're trying to physically destroy some resource rather than doing surgery on your statefile which is what `terraform state rm` is used for. Normally with Terraform you don't explicitly delete resources. Instead you change your configuration to no longer define a given resource, then next time you run `terraform apply` Terraform will automatically destroy the resource as it is no longer needed. Hopefully that helps, let me know if I misunderstood what you're trying to do.

**Comment by Lewiscowles1986** [comment]
Hi 👋  

I managed to make this work using action apply

I copied all of the code used by the module for the component part

```
module "app_name_queue" {
    source = "../../modules/business_aws_sqs"
```

so I copied `../../modules/business_aws_sqs` and added `count = 0` to all resources in the copied module file for copied resources (the module was just resources in a `sqs.tf` and a `variables.tf` file with `outputs.tf`.

```
module "app_name_queue" {
    source = "../../modules/business_aws_sqs_sunset"
```

This removed the IAM resources, SQS resources for a specific file within a component (module) by using a variant of the same module.

I think this speaks to components that are a little too broad. Instead of having one platform RDS for each of our applications, the business I work at has many within a single terraform (so destroy would delete a number of databases)

Thank you, I'll close this out
